examples/js/shaders/MMDToonShader.js

```diff
@@ -0,0 +1,110 @@
+( function () {
+
+	/**
+ * MMD Toon Shader
+ *
+ * This shader is extended from MashPhongMaterial, and merged algorithms with
+ * MashToonMaterial and MeshMetcapMaterial.
+ * Ideas came from https://github.com/mrdoob/three.js/issues/19609
+ *
+ * Combining steps:
+ *  * Declare matcap uniform.
+ *  * Add gradientmap_pars_fragment.
+ *  * Combine dotNL and gradient irradiances as total irradiance.
+ *    (Replace lights_phong_pars_fragment with lights_mmd_toon_pars_fragment)
+ *  * Add mmd_toon_matcap_fragment.
+ */
+	const lights_mmd_toon_pars_fragment = `
+varying vec3 vViewPosition;
+
+#ifndef FLAT_SHADED
+
+	varying vec3 vNormal;
+
+#endif
+
+
+struct BlinnPhongMaterial {
+
+	vec3 diffuseColor;
+	vec3 specularColor;
+	float specularShininess;
+	float specularStrength;
+
+};
+
+void RE_Direct_BlinnPhong( const in IncidentLight directLight, const in GeometricContext geometry, const in BlinnPhongMaterial material, inout ReflectedLight reflectedLight ) {
+
+	vec3 irradiance = getGradientIrradiance( geometry.normal, directLight.direction ) * directLight.color;
+	irradiance += saturate( dot( geometry.normal, directLight.direction ) ) * directLight.color;
+  irradiance *= 0.5;
```

Comment:
`RE_Direct_BlinnPhong` in r111 is

```
void RE_Direct_BlinnPhong( const in IncidentLight directLight, const in GeometricContext geometry, const in BlinnPhongMaterial material, inout ReflectedLight reflectedLight ) {
	#ifdef TOON
		vec3 irradiance = getGradientIrradiance( geometry.normal, directLight.direction ) * directLight.color;
	#else
		float dotNL = saturate( dot( geometry.normal, directLight.direction ) );
		vec3 irradiance = dotNL * directLight.color;
	#endif
	#ifndef PHYSICALLY_CORRECT_LIGHTS
		irradiance *= PI;
	#endif
	reflectedLight.directDiffuse += irradiance * BRDF_Diffuse_Lambert( material.diffuseColor );
	reflectedLight.directSpecular += irradiance * BRDF_Specular_BlinnPhong( directLight, geometry, material.specularColor, material.specularShininess ) * material.specularStrength;
}
```

For Toon material we didn't apply

```
		float dotNL = saturate( dot( geometry.normal, directLight.direction ) );
		vec3 irradiance = dotNL * directLight.color;
```

I think it may be the reason why the model with this PR looks brighter than the one in r111 with `MeshToonMaterial`.

Any specific reasons why you added this line?

```
irradiance += saturate( dot( geometry.normal, directLight.direction ) ) * directLight.color;
```

Reply:
Oh, good point!

I was thinking `dotNL` is the core idea of combining MeshToonMaterial and MeshPhongMaterial.

But now I found that what we actually need is just this part for specular after you point it out:
```
reflectedLight.directSpecular += irradiance * BRDF_Specular_BlinnPhong( directLight, geometry, material.specularColor, material.specularShininess ) * material.specularStrength;
```

I'll remove these two lines, many thanks!